remove  "boot from usb" kw

PLangowski opened the pull request.

PLangowski committed:
Remove usb boot kw

Signed-off-by: Pawel Langowski <pawel.langowski@3mdeb.com>

macpijan commented on dasharo-compatibility/coreboot-fan-control.robot:
```diff
@@ -32,7 +32,9 @@ CFN001.001 CPU temperature and fan speed can be read (Debian 11.02)
     Power On
-    Boot From USB
+    ${usb}=    Get USB Boot Option
+    Power On
```
Why do we need two power ons?

PLangowski replied:
` Boot System Or From Connected Disk ` assumes that the platform is rebooted (it launches the boot menu again). I could add a new keyword that boots the specified option from the boot menu without the need to restart. It would then make sense to wrap all of this in a single keyword, which would pretty much reintroduce the `Boot From USB` keyword. Is that okay?

macpijan replied:
It seems the idea in the original MR (https://github.com/Dasharo/open-source-firmware-validation/pull/92/files) was to use existing kwd for booting a menu entry, by simply using another argument. But it has not been completed as we can see. The assumption there was that the USB has static name defined in config IIUC, which should be fine in the most cases. We can also try to dynamically find this entry as you suggest. We should consider also what happens if we have multiple USB drives in this case.

PLangowski replied:
This is exactly what I'm doing here, but as of now it requires a restart, since the `Boot From ...` keyword launches the boot menu. I don't see how we can boot from USB without restarting if we don't want to create a new keyword.

macpijan reviewed (commented):
On which platforms it has been tested?
This issue needs a bit deeper understanding of why we would need booting from USB in these tests in the first place.
Which OSes are booted there, and are we really need them, instead of e.g. Ubuntu from disk, or DTS?

PLangowski commented:
@macpijan I tested this on MSI. I assumed that the tests needed USB boot when they used the keyword, but you're right.

The following tests used the `Boot from USB` option:
- Test cases in `dasharo-compatibility/usb-boot.robot`. These obv. need to boot from USB.
- `dasharo-compatibility/coreboot-fan-control.robot`: These may not need USB boot, as they just want to boot Debian.
-  `dasharo-compatibility/flash-write-protection.robot`: Same as above.

So perhaps in the last two suites the keywords should be replaced with ` Boot System Or From Connected Disk     debian`, what do you think?

macpijan commented:
>      So perhaps in the last two suites the keywords should be replaced with  Boot System Or From Connected Disk     debian, what do you think?

Sounds fine, maybe even Ubuntu as this is what we mostly test.
Which platform supports these tests? IIRC it could have been kgpe only. If this is the case, we might not care that much as it is not really supported platform.

PLangowski commented:
It seems that no platform config supports the fan control tests:
```
platform-configs/raptor-cs_talos2.robot
54:${COREBOOT_FAN_CONTROL_SUPPORT}=                    ${FALSE}

platform-configs/rpi-3b.robot
68:${COREBOOT_FAN_CONTROL_SUPPORT}=                    ${FALSE}

platform-configs/include/default.robot
92:${COREBOOT_FAN_CONTROL_SUPPORT}=                    ${FALSE}

platform-configs/qemu.robot
98:${COREBOOT_FAN_CONTROL_SUPPORT}=                    ${FALSE}

dasharo-compatibility/coreboot-fan-control.robot
24:...                     Skip If    not ${COREBOOT_FAN_CONTROL_SUPPORT}    coreboot fan control not supported
```

As for the write protection suite, it seems that PCEngines platforms support it

```
rg "HARDWARE_WP_SUPPORT"
dasharo-compatibility/flash-write-protection.robot
24:...                     Skip If    not ${HARDWARE_WP_SUPPORT}    Flash protection tests not supported

platform-configs/include/default.robot
123:${HARDWARE_WP_SUPPORT}=                             ${FALSE}

platform-configs/raptor-cs_talos2.robot
87:${HARDWARE_WP_SUPPORT}=                             ${FALSE}

platform-configs/include/pcengines.robot
69:${HARDWARE_WP_SUPPORT}=                     ${TRUE}

platform-configs/rpi-3b.robot
101:${HARDWARE_WP_SUPPORT}=                             ${FALSE}

platform-configs/qemu.robot
131:${HARDWARE_WP_SUPPORT}=                             ${FALSE}
```

I can change the OS to Ubuntu if you think that's a good idea.

m-iwanicki committed:
menus.robot: Get USB Boot Option - check if it's USB in Drive Path

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki reviewed (commented):
Please comment what do you think about my changes.

m-iwanicki committed:
dasharo-compatibility: usb-detect: Fix keywords

Remove non existing keywords and fix "Enter Tianocore" keyword

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
robot-compatibility: usb-{boot|detect}: Fail tests faster

No reason to try to boot 5 times if test will fail anyway after n-th
fail. Fail immediately after number of fails goes beyond threshold

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

macpijan reviewed (commented):
forgot to click `submit review` :facepalm:

PLangowski commented on dasharo-compatibility/coreboot-fan-control.robot:
```diff
@@ -30,21 +30,23 @@ Suite Teardown      Run Keyword
     [Documentation]    Check whether the data of CPU temperature and CPU fan
     ...    is available and can be read.
-    Skip If    not ${TESTS_IN_DEBIAN_SUPPORT}    CFN001.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    CFN001.001 not supported
```
https://github.com/Dasharo/open-source-firmware-validation/pull/312/commits/ae6a73d8fe7b792b3ba0796243ec1f9251c2bfbc#diff-18cd6de46364ae2c67ffd2e3dbde89ae3299f4e2708eaf83001ef97c18d69b2cR28

PLangowski commented on dasharo-compatibility/coreboot-fan-control.robot:
```diff
@@ -30,21 +30,23 @@ Suite Teardown      Run Keyword
     [Documentation]    Check whether CPU fan speed increases if the CPU
     ...    temperature rises.
-    Skip If    not ${TESTS_IN_DEBIAN_SUPPORT}    CFN002.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    CFN002.001 not supported
```
https://github.com/Dasharo/open-source-firmware-validation/pull/312/commits/ae6a73d8fe7b792b3ba0796243ec1f9251c2bfbc#diff-18cd6de46364ae2c67ffd2e3dbde89ae3299f4e2708eaf83001ef97c18d69b2cR40

macpijan commented:
Integrated in: https://github.com/Dasharo/open-source-firmware-validation/pull/659
